vpr/src/device/rr_graph_builder.h

```diff
@@ -40,6 +40,37 @@ class RRGraphBuilder {
     t_rr_graph_storage& node_storage();
     /** @brief Return a writable object for update the fast look-up of rr_node */
     RRSpatialLookup& node_lookup();
+
+    /** @brief Add a rr_segment to the routing resource graph. Return an valid id if successful.*/
+    inline RRSegmentId add_rr_segment(const t_segment_inf& segment_info) {
+        //Allocate an ID
+        RRSegmentId segment_id = RRSegmentId(segment_ids_.size());
+        segment_ids_.push_back(segment_id);
+
+        rr_segments_.push_back(segment_info);
+
+        return segment_id;
+    }
+    /** @brief Return a writable list of all the rr_segments */
```

Comment:
As suggested by @vaughnbetz , we should leave a TODO comment here:

/** @brief Return a writable list of all the rr_segments 
  * .. warning:: It is not recommended to use this API unless you have to. The API may be deprecated later, and future APIs will designed to return a specific data from the rr_segments.
  */